File: rust_kernel/src/taskmaster/syscall/getegid.rs
use super::scheduler::SCHEDULER;
use super::SysResult;

/// The getgid() function shall return the effective group ID of the
/// calling process. The getgid() function shall not modify errno.
pub fn sys_getegid() -> SysResult<u32> {
    Ok(unpreemptible_context!({
        SCHEDULER.lock().current_thread_group().credentials.egid
    }))
}

File: libc/src/unistd/getegid.c
#include <unistd.h>
#include <user_syscall.h>

// The getegid() function shall return the effective group ID of the
// calling process. The getegid() function shall not modify errno.

gid_t getegid(void)
{
	return (gid_t)_user_syscall(GETEGID, 0);
}
